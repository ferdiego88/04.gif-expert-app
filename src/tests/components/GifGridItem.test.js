import { shallow }from 'enzyme'
import React from 'react';
import GifGridItem from "../../components/GifGridItem";

describe('Expects <GifGridItem />', () => {

    const title = 'GifGridItem titulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow( <GifGridItem title={title} url={url}/>);


    test('should be to show <GifGridItem /> successfully', () => { 


        expect(wrapper).toMatchSnapshot();
     })

     test('should be a paragraph with the title', () => {

        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
     })

     test('should have a image and url equals to props', () => {

        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);

      })

      test('should be have animate__fadeIn', () => {

            const fadeIn = 'animate__fadeIn';
            const div = wrapper.find('div');
            
            expect(div.prop('className')).toContain(fadeIn);
       })
})