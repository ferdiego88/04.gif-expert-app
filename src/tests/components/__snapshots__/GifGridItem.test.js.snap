// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expects <GifGridItem /> should be to show <GifGridItem /> successfully 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="GifGridItem titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    GifGridItem titulo
  </p>
</div>
`;
